.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,139 @@
+---
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        type: string
+        default: self-hosted
+
+name: Commit lint and semantic release
+jobs:
+  setup:
+    name: Setup
+    runs-on: ${{ inputs.runs-on }}
+    outputs:
+      workflow_tag: ${{ steps.workflow_tag.outputs.workflow_tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Get workflow ref
+        id: workflow_tag
+        shell: bash
+        run: |
+          workflow_file="${GITHUB_WORKFLOW}"
+          if ! [ -f "${workflow_file}" ]
+          then
+            workflow_file="$(grep -l "^name: ${GITHUB_WORKFLOW}$" .github/workflows/*)"
+          fi
+          workflow_tag="$(awk -F @ '/github-workflow-semantic-release/ {print $2}' "${workflow_file}")"
+          echo "::set-output name=workflow_tag::${workflow_tag}"
+  commitlint:
+    needs: setup
+    name: Commit lint
+    runs-on: ${{ inputs.runs-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Checkout workflow
+        uses: actions/checkout@v3
+        with:
+          repository: nrkno/github-workflow-semantic-release
+          ref: ${{ needs.setup.outputs.workflow_tag }}
+          path: .github-workflow-semantic-release
+      - name: Set configuration
+        id: set_config
+        shell: bash
+        run: |
+          if [ -f "commitlint.config.js" ]
+          then
+            echo "Using callers configuration for commitlint"
+            echo "::set-output name=commitlintconfig::commitlint.config.js"
+          else
+            echo "Using workflow configuration for commitlint"
+            echo "::set-output name=commitlintconfig::.github-workflow-semantic-release/commitlint.config.js"
+          fi
+      - uses: wagoid/commitlint-github-action@v4
+        id: lint
+        continue-on-error: ${{ github.event_name == 'pull_request' }}
+        with:
+          configFile: ${{ steps.set_config.outputs.commitlintconfig }}
+      - uses: actions/github-script@v6
```

Comment:
Might be runner-specific whether we have Node.js installed or not. Could be smart to add an `setup-nodejs` job here.

Reply:
That is the responsibility of that action to install deps, not us.